### PR TITLE
Introduce error types in credentials api

### PR DESCRIPTION
**Pull request by Qubad786**

### CODE REVIEW

**Comment by Qubad786** [comment]
@muzaffaryousaf @muhammad-ammar @mushtaqak have a look.

File: VEDA_OS01/tests/test_views.py [hunk 1]
```diff
@@ -48,8 +50,6 @@ def test_transcript_credentials_unauthorized(self):
         )
 
         response_status_code = response.status_code
-        response = json.loads(response.content)
-
         self.assertEqual(response_status_code, status.HTTP_401_UNAUTHORIZED)
```
Comment (mushtaqak) [inline VEDA_OS01/tests/test_views.py]: I think we can use `response.status_code` in assert, but I guess it's not in your PR scope, was missed in previous PR I guess.

**Comment by mushtaqak** [comment]
Looks good :)

**Comment by Qubad786** [comment]
Going to merge.

**Pull request merged**